### PR TITLE
Drop melted update

### PR DESCRIPTION
**Pull request by apackin**
#### What's this PR do?
Refactors the melted user script to:
  search for melted users only inactive course
  track melted state using CTL online role
  update roles in CTL online correctly
Expand CTLonline API capabilities
Simplify some of removeChurned
#### Where should the reviewer start?
jobs/training/dropMeltedUsers.js
#### How should this be manually tested?
Create accounts in Canvas and CTL online with the same email address (I've just been using my own)
Modify account on CTLonline to be a trainee (have that 'Role')
Melt in a Canvas course (either by 'concluding Enrollment' https://crisistextline.instructure.com/courses/${courseID}/conclude_user/${userID} or Deactive the user in the course from GUI https://crisistextline.instructure.com/courses/${courseID}/users

Run script (Keys, Config, you know) 

'Role' in CTL online should be modified to 'Deferred' on melting from one course and 'Dropout' when melted for a second time. Emails should be sent. 

#### What are the relevant tickets?
https://admin.crisistextline.org/jira/secure/RapidBoard.jspa?rapidView=5&view=detail&selectedIssue=INT-472
#### Questions:
Bluebird is already required by the request module. Do we need to require it again in the main package.json?

### CODE REVIEW

File: jobs/scheduling/RecurShifts.js [hunk 1]
```diff
@@ -33,15 +33,15 @@ function recurNewlyCreatedShifts() {
...
 
   WhenIWork.get('shifts', postData, function(response) {
-    const usersWithNewShifts = response.users;
-    var allShifts = response.shifts;
-    if (typeof allShifts !== 'object') {
+    if (!response || !response.shifts || typeof response.shifts !== 'object') {
       CONSOLE_WITH_TIME('NO SHIFTS RETURNED.');
       CONSOLE_WITH_TIME('===================');
       CONSOLE_WITH_TIME('RESPONSE: ' + response);
       CONSOLE_WITH_TIME('POST: ' + postData);
```
Comment (mbogor) [inline jobs/scheduling/RecurShifts.js]: use \n and do it in one console

File: jobs/training/graduateCanvasUserToPlatform.js [hunk 2]
```diff
@@ -15,21 +17,15 @@ throttler.configure({
   milliseconds: 1000
 });
 
-new CronJob(CONFIG.time_interval.graduate_users_cron_job_string, function () {
-    pollCanvasForGraduatedUsersThenCreatePlatformAccount();
-  }, null, true);
+new CronJob(CONFIG.time_interval.graduate_users_cron_job_string, graduateCanvasUserToPlatform, null, true);
```
Comment (mbogor) [inline jobs/training/graduateCanvasUserToPlatform.js]: bad name

File: jobs/training/meltedDataDumpToCSV.js [hunk 3]
```diff
@@ -6,7 +6,7 @@
 // const CronJob = require('cron').CronJob;
```
Comment (tongxiang) [inline jobs/training/meltedDataDumpToCSV.js]: Can you add a comment about why we're commenting this out, for now? 

(i.e., this isn't useful right now, or we're not making data dumps on Elmer right now, etc.)

File: jobs/training/meltedDataDumpToSlack.js [hunk 4]
```diff
@@ -8,7 +8,7 @@
 // const internalRequest = require('request');
```
Comment (tongxiang) [inline jobs/training/meltedDataDumpToSlack.js]: same

**Comment by tongxiang** [comment]
Yes, bluebird still needs to be required in main package.json. (Since request as a module doesn't expose bluebird to the global namespace for our app.)

**Comment by tongxiang** [comment]
👍 good, easy-to-read, intuitive refactor.

File: jobs/training/meltedDataDumpToCSV.js (hunk 3 above)
Comment (apackin) [inline jobs/training/meltedDataDumpToCSV.js]: These were commented out in a previous commit when Elmer was failing unexpectedly. I will add them back right now in a separate pr.

**Pull request merged**